core/grafana-data-source/src/util.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+/// Get the current unix timestamp in milliseconds.
+pub fn now_millis() -> i64 {
+	chrono::Utc::now().timestamp_millis()
```

Comment:
Could use `SystemTime` so that we don't depend on `chrono`:

Something like that (pseudo-code, needs to handle negative values):

```suggestion
	let dur = SystemTime::UNIX_EPOCH.elapsed().unwrap(); dur.as_secs().saturating_mul(1_000_000).saturating_add(dur.subsec_nanos() / 1000)
```

Reply:
I'm not sure we can get rid of `chrono` without replacing it with something else because we need a way to parse the timestamp strings coming from grafana.